operations/observability/Observability.md

```diff
@@ -0,0 +1,336 @@
+  - Stage duration (the difference between the stage end time and stage start time)
+  - Stage status (status is marked as succeeded if all the operations' statuses are succeeded, otherwise it is marked as failed)
+- Product timing metrics
+  - Product name
+  - Product start time (start time of the process-media stage)
+  - Product end time (start time of the post-install-check stage)
+  - Product status (status is marked as succeeded if all the stages' statuses are succeeded, otherwise it is marked as failed)
+
+## Goss tests for PIT and NCN
+
+### Goss test overview
+
+The observability tooling monitors the logs for Goss tests run by using the automated scripts. These scripts are regularly run during install/upgrade. For each Goss test, metrics are generated on its duration and success or failure.
+
+This framework provides a set of quantifiable metrics used to create a visual Grafana health dashboard of all environments. This provides administrators with insight on which areas are experiencing problems.
+
+Trend analysis of this data in Grafana across different dimensions may point out statistically where the highest frequency of issues occurs.
+It also provides administrators an at-a-glance dashboard where they can visually see the complete system health. This includes the option to drill down from the aggregate view of the environment to the functional areas or the components.
+
+Goss test logs are scraped using grok-exporter and visualized on Grafana using captured data.
+
+Goss tests are contained in YAML file. A YAML file containing a list of Goss test files is called a test suite. These suites are run through the CSM-provided scripts to get a log file with results for the tests.
+
+### Workflow
+
+![Goss test workflow](../../img/operations/GossWorkflow.png "Goss Workflow")
+
+### Log file format
+
+Individual lines of logs are in the following format for each node, test name, and source:
+
+```json
+{
+  "log_timestamp": "20230118_094205.821955",
+  "Product": "CSM",
+  "log_script": "print_goss_json_results.py",
+  "log_message": "Test result",
+  "Description": "Validates that 'cray --version' is available and executes successfully on the local system.",
+  "Execution Time (nanoseconds)": 1312368478,
+  "Execution Time (seconds)": 1.312368478,
+  "Node": "ncn-m001",
+  "Result Code": 0,
+  "Result String": "PASS",
+  "Source": "http://ncn-m001.hmn:8997/ncn-healthcheck-master",
+  "Test Name": "Command 'cray --version' Available",
+  "Test Summary": "Command: command_available_1_cray_version: exit-status: matches expectation: [0]"
+}
+```
+
+### Grok-exporter deployment, service, and service-monitor
+
+Grok-exporter is deployed on all of the Kubernetes master nodes, by using node affinity in its deployment.
+As background information, the service for the grok-exporter is accessible at port `9144` and service-monitor implementation is for Prometheus to access the metrics that are created by the grok-exporter.
```

Comment:
Is ````service-monitor``` related to the grok-exporter accessible at port 9144?  I am a little confused how these two points in this sentence are related.